### PR TITLE
Add outbound policy e2e test

### PR DESCRIPTION
**Pull request by adleong**
Add an e2e test for outbound HTTPRoute based routing.

### CODE REVIEW

File: policy-test/src/curl.rs [hunk 1]
```diff
@@ -181,7 +181,7 @@ impl Runner {
                     name: "curl".to_string(),
                     image: Some("docker.io/curlimages/curl:latest".to_string()),
                     args: Some(
-                        vec!["curl", "-sf", "-o", "/dev/null", "-w", "%{http_code}", "--max-time", "10", "--retry", "12", target_url]
+                        vec!["curl", "-sfv", "-o", "/dev/null", "-w", "%{http_code}", "--max-time", "10", "--retry", "10", "--retry-delay", "2", target_url]
```
Comment (adleong) [inline policy-test/src/curl.rs:184]: If `--retry-delay` is not specified, the time between each subsequent try doubles.  This means that retries go on for a VERY long time and 5XX errors from downstream services are not exposed.

**Review by mateiidavid — APPROVED** [review]

File: policy-test/tests/e2e_http_routing.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                    inner: k8s::policy::httproute::CommonRouteSpec {
+                        parent_refs: Some(vec![k8s::policy::httproute::ParentReference {
+                            namespace: None,
+                            name: "web".to_string(),
+                            port: Some(80),
+                            group: Some("core".to_string()),
+                            kind: Some("Service".to_string()),
+                            section_name: None,
+                        }]),
+                        ..Default::default()
```
Comment (mateiidavid) [inline policy-test/tests/e2e_http_routing.rs:25]: nit: might be unused according to clippy

File: policy-test/src/web.rs [hunk 3]
```diff
@@ -25,6 +25,7 @@ pub fn pod(ns: &str) -> k8s::Pod {
                     container_port: 8080,
                     ..Default::default()
                 }]),
+
```
Comment (alpeb) [inline policy-test/src/web.rs:28]:
```suggestion
```

**Review by alpeb — APPROVED** [review]
LGTM, module clippy warning :+1:

**Pull request merged**